### DISCUSSION TITLE
Allowed values of function parameters

### DISCUSSION THREAD

**Discussion by AlexStox** [highfestiva/finplot#148] [Q&A]
Hi Highfestiva, how can I get possible values for one of the function parameters? 

For instance, I'm curious about finplot.plot's style parameter - what values are possible for style? Is it style for the entire chart, like bar, ohlc, line? And what parameter can change a style of a chart?

Where can I read more about every method of finplot?

**Comment by highfestiva** [comment]
`plot()` can do line plots and scatter plots. Scatter style can be `o`, `^`, `v`, `>`, `<` and then some. Line style can be a combination of `-` (short dash), `_` (underscore means long dash), `.` (dot) and ` ` (space). If you instead do `df.plot(kind='histogram')` you mean instead what category of plot. All of this is in source, and most of it in examples. So check them out, and see finplot's source for more info.

**Reply by AlexStox** [reply]
Well, now I got it - style in plot() is for scatter plot. I used your major example to plot a chart.
```
import finplot as fplt
start_date = '2010-01-01'
end_date = '2020-10-30'
ax,ax2 = fplt.create_plot("SPX", rows=2)
df_show = df[['Open','Close','High','Low','TrueRange']]
fplt.candlestick_ochl(df_show.loc[start_date:end_date,:], ax=ax)
```

This creates a chart on the 2nd row of the plot window, but as I understood it's not possible to set the bar style
`fplt.plot(df_show.index, y=df_show.loc[:,'TrueRange'].to_numpy(), ax=ax2, color='#927', legend='TrueRange')`
So, now I got a point - `.plot()` is not the solution to plot everything and see on a chart, I should use other methods to tune the chart the way I want, e.g. `fplt.candlestick_ochl()` makes a candle chart. 

So I tried this method
`fplt.bar(x=df_show.loc[start_date:end_date,'TrueRange'], ax=ax2)`
and realized what the description means in `.bar()`: _'''Bar plots are decoupled. Use volume_ocv() if you want a bar plot which relates to other time plots.'''__ 
    -> when we use `.bar()` for some data, this data will not be related to other plotted data on the chart? For instance, if I plotted on 1st row the OHLC candlestick chart, and would like to plot "TrueRange" data related to the 1st chart ("SPX"), they won't be 'linked' (chained) by time, won't be related? Because when I zoomed or drag the 1st chart, the 2nd chart doesn't change.
2. **And for such purposes you recommend using**  `volume_ocv()` ?
I tried to use it like
`fplt.volume_ocv(df_show.loc[start_date:end_date,:], ax=ax2)`
but it didn't work. 

3. Can you please advise on how to plot "TrueRange" column data of my df_show dataframe as the 2nd row (chart) on the plot (where 1st row (chart) is "SPX" OHLC candlestick chart?

4. Do you recommend resetting df DatetimeIndex to use finplot? 
(<index:0,1,2,...> <column1: dates> <column2: Open> ...)
Because in the code below and somewhere in description I see that DatetimeIndex may not be count.  
```
strength_colorfilter(item, datasrc, df):
    opencol = df.columns[1]
    closecol = df.columns[2]
    startcol = df.columns[3]
    endcol = df.columns[4]
```

**Reply by highfestiva** [reply]
2. See examples. You want three columns for volume: open, close and volume. The time can either be the first column or the index, it doesn't matter. If you use timestamps on the first column, make sure to `df.reset_index(drop=True)` to make sure you're using unique, zero-based and performant indexes.
3. Not sure what you mean without an example, but if you have datetime in the index something like `fplt.plot(df_show['TrueRange'], ax=ax2)`?
4. Nope, I recommend using datetime in the index as long as you're plotting time series. See yfinance's eminent format for instance.

**Reply by AlexStox** [reply]
well, `fplt.plot(x=df_show.loc[start_date:end_date,'TrueRange'], ax=ax2)` plots it as a line
![image](https://user-images.githubusercontent.com/55058601/112385229-08a8f280-8cf8-11eb-9d91-2624e5160a64.png)

how to plot it as a bar chart?

**Reply by highfestiva** [reply]
```python
fplt.volume_ocv(df_show.loc[start_date:end_date, ['Open', 'Close', 'TrueRange']], ax=ax2)
```
might work?

**Reply by AlexStox** [reply]
great, thanks, it worked :) I'll play with it later and reply if have more questions :) Thank you!

**Reply by AlexStox** [reply]
I couldn't get whether it possible to change colorfunc in volume_ocv? E.g. if I need my indicator to be single-colored like in blue.

**Reply by highfestiva** [reply]
It checks if open>=close to assign bull or bear colors. Use something like
```python
df['tr-open'] = 0
df['tr-close'] = 1
fplt.volume_ocv(df_show.loc[start_date:end_date, ['tr-open', 'tr-close', 'TrueRange']], ax=ax2)
```
to get all bullish color. Then adjust color of plot afterwards (as shown in some example) if you want something specific.
